### ISSUE TITLE
Illegal reflective access operation with Java 11 and registering a Resource instance

### ISSUE THREAD

**Issue by pstackle** [dropwizard/dropwizard#2962]
Using Dropwizard v2.0.0-rc10 and Java 11.0.4 I get the following warning when I register an already constructed instance of a resource class.

```
environment.jersey().register(new ProtectedResource());
```

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by javassist.util.proxy.SecurityActions (file:/Users/username/.m2/repository/org/javassist/javassist/3.25.0-GA/javassist-3.25.0-GA.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of javassist.util.proxy.SecurityActions
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

If I run with the `--illegal-access=deny` option configured, then I receive the following error:

```
java.lang.RuntimeException: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @60c6f5b

  at io.dropwizard.jersey.DropwizardResourceConfig.register(DropwizardResourceConfig.java:160)
  at io.dropwizard.jersey.setup.JerseyEnvironment.register(JerseyEnvironment.java:36)
  at com.apple.amp.auth.cert.dropwizard.test.ProtectedApp.run(ProtectedApp.java:36)
  at com.apple.amp.auth.cert.dropwizard.test.ProtectedApp.run(ProtectedApp.java:22)
  at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:43)
  at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
  at io.dropwizard.testing.DropwizardTestSupport.startIfRequired(DropwizardTestSupport.java:315)
  at io.dropwizard.testing.DropwizardTestSupport.before(DropwizardTestSupport.java:208)
  at io.dropwizard.testing.junit.DropwizardAppRule.before(DropwizardAppRule.java:213)
  at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
  at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
  at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
  at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
  at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
  at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @60c6f5b
  at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:340)
  at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:280)
  at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:198)
  at java.base/java.lang.reflect.Method.setAccessible(Method.java:192)
  at javassist.util.proxy.SecurityActions.setAccessible(SecurityActions.java:159)
  at javassist.util.proxy.DefineClassHelper$JavaOther.defineClass(DefineClassHelper.java:213)
  at javassist.util.proxy.DefineClassHelper$Java11.defineClass(DefineClassHelper.java:52)
  at javassist.util.proxy.DefineClassHelper.toClass(DefineClassHelper.java:260)
  at javassist.ClassPool.toClass(ClassPool.java:1232)
  at javassist.ClassPool.toClass(ClassPool.java:1090)
  at javassist.ClassPool.toClass(ClassPool.java:1048)
  at javassist.CtClass.toClass(CtClass.java:1290)
  at io.dropwizard.jersey.DropwizardResourceConfig.register(DropwizardResourceConfig.java:156)
  ... 16 more
```

This seems to be an issue with the `DropwizardResourceConfig.register(Object)` method calling `CtClass.toClass()`.

```
CtClass cc = pool.makeClass(SpecificBinder.class.getName() + UUID.randomUUID());
cc.setSuperclass(pool.get(SpecificBinder.class.getName()));
Object binderProxy = cc.toClass().getConstructor(Object.class, Class.class).newInstance(object, clazz);
```

In `CtClass.toClass()`, it has the following javadoc snippet:
```
     * <p><b>Warning:</b> In Java 11 or later, the call to this method will
     * print a warning message:</p>
     * <blockquote><pre>
     * WARNING: An illegal reflective access operation has occurred
     * WARNING: Illegal reflective access by javassist.util.proxy.SecurityActions$3 ...
     * WARNING: Please consider reporting this to the maintainers of javassist.util.proxy.SecurityActions$3
     * WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
     * WARNING: All illegal access operations will be denied in a future release
     * </pre></blockquote>
     * <p>To avoid this message, use {@link #toClass(Class)}
     * or {@link #toClass(java.lang.invoke.MethodHandles.Lookup)}.
     * {@link #toClass()} will be unavailable in a future release.
     * </p>
```

Java version info:

```
java -version
openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.4+11, mixed mode)
```

PR with a proposed fix to follow.

**Comment by pstackle** [comment]
Unfortunately, the suggested alternatives to `toClass()` are not compatible with Java 8.

**Comment by joschi** [comment]
@pstackle I think I've found a way to fix it with minimal changes in #2964.
